x-pack/plugins/endpoint/server/routes/alerts.ts

```diff
@@ -5,27 +5,101 @@
  */
 
 import { IRouter } from 'kibana/server';
+import { RequestHandler } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema, TypeOf } from '@kbn/config-schema';
 
-import json from './sampledata.json';
+import {
+  getPagingProperties,
+  kibanaRequestToAlertListQuery,
+} from '../services/endpoint/alert_query_builders';
+
+import { AlertData, AlertResultList } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const ALERTS_ROUTE = '/api/endpoint/alerts';
+
+export const reqSchema = schema.object({
+  page_size: schema.number({ defaultValue: 10, min: 1, max: 10000 }),
+  page_index: schema.number({ defaultValue: 0, min: 0 }),
+});
+
+export function registerAlertRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const alertsHandler: RequestHandler<unknown, TypeOf<typeof reqSchema>> = async (
+    ctx,
+    req,
+    res
+  ) => {
+    try {
+      const queryParams = await getPagingProperties(req, endpointAppContext);
+      const reqBody = await kibanaRequestToAlertListQuery(queryParams, endpointAppContext);
+      const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser(
+        'search',
+        reqBody
+      )) as SearchResponse<AlertData>;
+      return res.ok({ body: mapToAlertResultList(endpointAppContext, queryParams, response) });
+    } catch (err) {
+      return res.internalError({ body: err });
+    }
+  };
 
-export function registerAlertRoutes(router: IRouter) {
   router.get(
     {
-      path: '/api/endpoint/alerts',
-      validate: false,
+      path: ALERTS_ROUTE,
+      validate: {
+        query: reqSchema,
+      },
       options: { authRequired: true },
     },
-    async (context, req, res) => {
-      try {
-        return res.ok({
-          body: json,
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        });
-      } catch (err) {
-        return res.internalError({ body: err });
-      }
-    }
+    alertsHandler
   );
+
+  router.post(
+    {
+      path: ALERTS_ROUTE,
+      validate: {
+        body: reqSchema,
+      },
+      options: { authRequired: true },
+    },
+    alertsHandler
+  );
+}
+
+function mapToAlertResultList(
+  endpointAppContext: EndpointAppContext,
+  queryParams: Record<string, any>,
```

Comment:
Could we add a type for these queryParams?

Reply:
Because this is the type used in elastic search, which is string to any type. Because the query params can take any shape depending on the query. It will be futile to type this any stronger. 

https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/scoped_cluster_client.ts#L67